Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/NodeTest.java

```diff
@@ -0,0 +1,1134 @@
+
+            Node condition = storage.createNode(Mods.NAME).put(RDF.toURL(9),
+                storage.createNode("http://names.example/alice"));
+
+            assertThat(node.matches(condition), is(false));
+        }
+    }
+
+    /**
+     * Tests {@code putAll(IdentifiableNode, Collection<? extends ObjectType>)}.
+     */
+    @Test
+    public void testPutAllIdentifiableNodeSet() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            List<ObjectType> nodes = new ArrayList<>();
+            nodes.add(storage.createNodeReference("http://names.example/alice"));
+            nodes.add(storage.createNodeReference("http://names.example/bob"));
+            nodes.add(storage.createNodeReference("http://names.example/charlie"));
+
+            Node node = storage.createNode();
+            node.putAll(Mods.NAME, nodes);
+
+            assertThat(node.get(Mods.NAME).countUntil(Long.MAX_VALUE), is((long) 3));
+        }
+    }
+
+    /**
+     * Tests {@code putAll(String, Collection<? extends ObjectType>)}.
+     */
+    @Test
+    public void testPutAllStringCollection() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            List<ObjectType> nodes = new ArrayList<>();
+            nodes.add(storage.createNodeReference("http://names.example/alice"));
+            nodes.add(storage.createNodeReference("http://names.example/bob"));
+            nodes.add(storage.createNodeReference("http://names.example/charlie"));
+
+            Node node = storage.createNode();
+            node.putAll(Mods.NAME.getIdentifier(), nodes);
+
+            assertThat(node.get(Mods.NAME).countUntil(Long.MAX_VALUE), is((long) 3));
+        }
+    }
+
+    /**
+     * Tests {@code put(IdentifiableNode, ObjectType)}.
+     */
+    @Test
+    public void testPutNodeReferenceObjectType() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNode().put(Mods.NAME, storage.createLeaf("Wilhelm Busch", "de"));
+        }
+    }
+
+    /**
+     * Tests {@code put(IdentifiableNode, String)}.
+     */
+    @Test
+    public void testPutNodeReferenceString() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNode().put(Mods.NAME, "Wilhelm Busch");
+        }
+    }
+
+    /**
+     * Tests {@code put(String, ObjectType)}.
+     */
+    @Test
+    public void testPutStringObjectType() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNode().put("http://www.loc.gov/mods/v3#name", storage.createLeaf("Wilhelm Busch", "de"));
+        }
+    }
+
+    /**
+     * Tests {@code put(String, String)}.
+     */
+    @Test
+    public void testPutStringString() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNode().put("http://www.loc.gov/mods/v3#name", "Wilhelm Busch");
```

Comment:
What did you testing here?

Reply:
Tests that put method with two strings as arguments does not throw exception